cmd/soroban-rpc/internal/config/options.go

```diff
@@ -209,6 +209,15 @@ func (cfg *Config) options() ConfigOptions {
 			ConfigKey:    &cfg.CheckpointFrequency,
 			DefaultValue: uint32(64),
 		},
+		{
+			Name: "ledger-retention-window",
```

Comment:
Is the intention here that this be a new, single retention-window to rule them all? Or are you introducing a new/different retention window just for ledgers (ie. distinct from `event-retention-window` and `transaction-retention-window`)? What we want to do is the former, ie create a single retention window to rule them all. In that vein, I would rename this to something more generic, maybe `history-retention-window` or directly mimic horizon's config (`history-retention-count`).

Reply:
Yes, we will eventually have only one retention window `history-retention-window` so makes sense what you said in both the comments. Refactored !!